src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,448 @@
+        :return boolean flag; True if archive is properly extracted
+        """
+        if path_to_extract is None:
+            path_to_extract = os.path.dirname(self.path)
+
+        _, *extension = self.path.split(".")
+        if extension != ["tar", "gz"]:
+            raise ValueError(
+                "Attempting to extract tar archive, but the extension "
+                f"of the Directory object is {extension}, not `.tar.gz`"
+            )
+
+        tar = tarfile.open(self.path, "r")
+        tar.extractall(path=path_to_extract)
+        tar.close()
+        return True
+
+    def __len__(self):
+        return len(self.files)
+
+    def _infer_path_from_files(self) -> Union[str, None]:
+        # if Directory object describes a local directory, we may try to infer the path
+        # from the attributes of the files contained within
+        paths = [os.path.dirname(file.path) for file in self.files]
+
+        # assert that all the files have the same dirname.
+        if len(set(paths)) == 1:
+            return paths[0]
+        else:
+            return None
+
+
+class FrameworkFiles(Directory):
+    """
+    Object that represents a directory with framework files.
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        """
+        Constructor for a FrameworkFiles object.
+
+        :param files: list of files contained within the FrameworkFiles
+        :param name: name of the FrameworkFiles
+        :param path: path of the FrameworkFiles
+        :param url: url of the FrameworkFiles
+        """
+
+        super().__init__(files=files, name=name, path=path, url=url)
+
+        self.nested_folders_names = ["checkpoint_(.*)/", "logs/"]
+
+    # TODO: Add support for model cards (pull integration from
+    #  model.md and specify which files to validate)
+    def validate(self, integration: Optional[str] = None) -> bool:
+        """
+        Validate whether all the framework files adhere
+        to the convention imposed by the integration.
+
+        :param integration: integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :return: boolean flag; True if files are valid and no errors arise
+        """
+        for file in self.files:
+            if isinstance(file, Directory):
+                # check for Directory files
+                self._check_directory(directory=file)
+            else:
+                # check for File files
+                self._check_file(file=file)
+
+        return True
+
+    def get_file(self, file_name: str) -> Optional[File]:
+        """
+        Fetch a file from the FrameworkFiles by name.
+
+        :param file_name: name of the file to be fetched
+        :return: File if found, otherwise None
+        """
+        for file in self.files:
+            if isinstance(file, Directory):
```

Comment:
should also check if directory name matches and return the directory if so

Reply:
Good catch